File: src/main/java/spring/webmvc/presentation/dto/request/ItemSaveRequest.java
package spring.webmvc.presentation.dto.request;

import jakarta.validation.constraints.Max;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import spring.webmvc.domain.model.enums.Category;

public record ItemSaveRequest(
	@NotBlank
	String name,
	@NotBlank
	String description,
	@Min(100)
	int price,
	@Max(9999)
	int quantity,
	@NotNull
	Category category
) {
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/presentation/dto/request/ItemSaveRequest.kt
package spring.web.kotlin.presentation.dto.request

import jakarta.validation.constraints.Max
import jakarta.validation.constraints.Min
import jakarta.validation.constraints.NotBlank
import spring.web.kotlin.domain.model.enums.Category


data class ItemSaveRequest(
    @field:NotBlank
    val name: String,
    @field:NotBlank
    val description: String,
    @field:Min(100)
    val price: Int,
    @field:Max(9999)
    val quantity: Int,
    val category: Category,
)
